### DISCUSSION TITLE
Navidrome should use relative (from the MusicFolder) path for writing track location in M3U files

### DISCUSSION THREAD

**Discussion by ghost** [navidrome/navidrome#2351] [Ideas]
I think Navidrome should use relative (from the MusicFolder) rather than absolute paths for its playlist creation. Current behaviour breaks all playlists if the root path is altered (e.g. new system/new disk/mountpoint or moving your collection to a different path). If path is set relative to the MusicFolder variable then playlists will continue to function should you ever need to move your collection. With relative path you would not need to edit hundreds of M3U files - all you would need to do is update the MusicFolder variable.

**Comment by deluan** [comment]
I think this is a good idea, but will not prevent your library to break if you change the MusicFolder config. Currently if you do this, you will loose all love/ratings/bookmarks you may have. Also, all playlists will be empty after the change, as the DB stores absolute paths for all songs. I want to change this when introducing multiple music folders (#192), and will also take a look at the M3U generation logic.

**Comment by wkelton** [comment]
I would love Navidrome to support file paths in playlists being relative to the music folder. I am managing my playlists with beets, ~~which does this (and doesn't support relative to the plyalist file like Navidrome). So, my source playlist files are outside Navidrome's music folder and I copy modified playlists inside Navidrome's music folder (with '../' prepended to each path).~~

EDIT: beets playlist plugin has a configuration that allows it to manage the playlists with files relative to the playlists dir (`relative_to: /music/playlists`) like Navidrome.
